src/src/PluginStructs/P122_data_struct.cpp

```diff
@@ -0,0 +1,568 @@
+        }
+        else
+        {
+          _last_action_started = millis();
+          _state = P122_state::New_Values_Available;
+          stable = true;
+        }
+      }
+    break;
+
+    case P122_state::Error:
+    case P122_state::New_Values_Available:
+      //this state is used outside so all we need is to stay here
+      stable = true;
+    break;
+
+    default: //anything not defined above sends the device for initialization
+      _state = P122_state::Uninitialized;
+  } // switch
+
+  if (_state != oldState)
+  {
+    if (loglevelActiveFor(LOG_LEVEL_INFO)) 
+    {
+      String log = F("SHT2x : ***state transition ");
+      log += String((int)oldState);
+      log += F("-->");
+      log += String((int)_state);
+      addLog(LOG_LEVEL_INFO, log);
+    }
+  }
+  return stable;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Returns the I2C connection state
+// Note: based upon the FSM state without actual accessing the device
+bool P122_data_struct::isConnected()
+{
+  switch (_state)
+  {
+    case  P122_state::Initialized:
+    case  P122_state::Ready:
+    case  P122_state::Wait_for_temperature_samples:
+    case  P122_state::Wait_for_humidity_samples:
+    case  P122_state::New_Values_Available:
+      return true;
+      break;
+    case  P122_state::Uninitialized:
+    case  P122_state::Error:
+    case  P122_state::Wait_for_reset:
+    default:
+      return false;
+      break;
+  }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Returns if the device communication is in error
+// Note: based upon the FSM state without actual accessing the device
+bool P122_data_struct::inError()
+{
+  return _state == P122_state::Error;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Returns if new acquired values are available
+bool P122_data_struct::newValues()
+{
+  return _state == P122_state::New_Values_Available;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Restart the FSM used to access the device
+bool P122_data_struct::reset()
+{
+  _state = P122_state::Uninitialized;
+  return true;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Start a new measurement cycle
+bool P122_data_struct::startMeasurements()
+{
+  if ((_state == P122_state::New_Values_Available) || (_state == P122_state::Initialized))
+  {
+    _state = P122_state::Ready;
+  }
+  return true;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Get the electronic idenfification data store in the device
+// Note: The data is read from the device during initialization 
+bool P122_data_struct::getEID(uint32_t *eida, uint32_t *eidb, uint8_t *firmware)
```

Comment:
Why not use references here instead of pointers?
```
bool P122_data_struct::getEID(uint32_t &eida, uint32_t &eidb, uint8_t &firmware)
```

Reply:
Uh, old skool C programmer...